### DISCUSSION TITLE
V8: how is custom paths supposed to work?

### DISCUSSION THREAD

**Discussion by niklasfjeldberg** [nuxt-modules/i18n#1691] [Q&A] [answered]
I am trying out the latest version of V8 and I wonder if this is how custom paths are supposed to work, or if there is a bug. Appreciate any advice.

Default language is `no`, while backup is `en`. And I have the file path `/pages/recipes/index.vue` for the route I am trying out.

```
lazy: false,
langDir: 'data/locales/',
defaultLocale: 'no',
vueI18n: {
  fallbackLocale: 'en',
  legacy: false
},
strategy: 'prefix_and_default',
customRoutes: 'config',
pages: {
  'recipes/index': {
    en: '/recipes',
    no: '/oppskrifter'
  },
  etc...
}
```

The following is the result when using `localePath` in a component.
```
localePath('/recipes') -> /recipes
localePath('/recipes', 'no') -> /recipes
localePath('/recipes', 'en') -> /en/recipes
```

I am expecting that `localePath('/recipes')` will have `/oppskrifter` as output.

**Comment by kazupon** [comment, marked as answer]
v8 custom paths are implemented in the same spec as v7. As mentioned in this issue.
unfortunately, it only works with named routes.
https://github.com/nuxt-modules/i18n/issues/641

```ts
localePath({ name: 'recipes' }) -> /en/recipes
localePath({ name: 'recipes' }, 'no') -> /oppskrifter
localePath({ name: 'recipes' }, 'en') -> /en/recipes
```

**Reply by aligzl** [reply]
hi @kazupon 

How to make it work? i cant make it work with dynamic parts.

Link: [Reproduction](https://stackblitz.com/edit/github-w7smfy-qnf45n?file=nuxt.config.ts,pages%2Findex.vue)

Thanks!

**Reply by BobbieGoede** [reply]
@aligzl 

Your usage of `localePath` like this is what you're looking for:
```ts
localePath({
  name: 'karriere-standorte-detail-locationname',
  params: { locationname: 'yourParamHere' },
})
```

For me it helps to check devtools > routes if I'm not sure what the route name ended up being

![image](https://user-images.githubusercontent.com/6649305/225255516-d30a46b3-135d-4a1d-b4c0-817f76deff1a.png)

If you omit the `___xx` at the end of the route name you have the name you want to use.

**Reply by aligzl** [reply]
hi @kazupon 

Thanks for your reply. It works now. I think it's very good idea to add this usage to the documentation.

Thanks.
